### PR TITLE
Lps 109995 master

### PR DESCRIPTION
**Pull request by rolandpakai**
Hi @dacousalr 

could you please check my changes?
Unortunatelly it is still not perfect: clicking on an item in the ItemSelectorDialog not calls the selectedItemChange event. The event is only called when the Dialog is closed.

Thanks, Roland

### CODE REVIEW

**Comment by dacousalr** [comment]
Just started reviewing :)

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/processors/getAlloyEditorProcessor.js [hunk 1]
```diff
@@ -68,6 +69,31 @@ export default function getAlloyEditorProcessor(
 			_editor = AlloyEditor.editable(getEditorWrapper(element), {
 				...editorConfig,
 
+				documentBrowseLinkCallback: (editor, url, changeLinkCallback) => {
+				
+					const itemSelectorDialog = new ItemSelectorDialog({
+						eventName: editor.name + 'selectDocument',
```
Comment (dacousalr) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/processors/getAlloyEditorProcessor.js]: Wouldn't this be better like `eventName: editorName + 'selectItem',` ?

File: modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/editor/configuration/FragmentEntryLinkEditorConfigContributor.java [hunk 2]
```diff
@@ -72,6 +72,8 @@ public void populateConfigJSONObject(
 			"allowedContent", ""
 		).put(
 			"disallowedContent", "br"
+		).put(
+			"documentBrowseLinkUrl", itemSelectorURL.toString()
```
Comment (dacousalr) [inline modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/editor/configuration/FragmentEntryLinkEditorConfigContributor.java]: Is this really needed?

**Comment by dacousalr** [comment]
Hi @rolandpakai,

I have added a couple of comments. Please, take a look at them.

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/processors/getAlloyEditorProcessor.js (hunk 1 above)
Comment (rolandpakai) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/processors/getAlloyEditorProcessor.js]: Yes, we can change it to selectItem

File: modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/editor/configuration/FragmentEntryLinkEditorConfigContributor.java (hunk 2 above)
Comment (rolandpakai) [reply modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/editor/configuration/FragmentEntryLinkEditorConfigContributor.java]: Yes, we need it at the frontedn side here: 
https://github.com/liferay/alloy-editor/blob/master/src/components/buttons/button-link-edit-browse.jsx#L76

**Comment by rolandpakai** [comment]
Thanks @dacousalr 
i added my comments too.

File: modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/editor/configuration/FragmentEntryLinkEditorConfigContributor.java (hunk 2 above)
Comment (dacousalr) [reply modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/editor/configuration/FragmentEntryLinkEditorConfigContributor.java]: Yes, I am aware of that.

My question was why these *browseLinkUrl's are needed if no toolbars are shown. The answer seems to be that there are not toolbars defined: 
`jsonObject.put("toolbars", JSONFactoryUtil.createJSONObject());`

I think that these urls were blind-copied in 3be60014db939f6fbd23bbf542de9a3340702066, so they can be removed.

**Comment by dacousalr** [comment]
Hi Roland,

Please, resend this PR making the changes based on our conclusions.

**Comment by rolandpakai** [comment]
Hi @dacousalr ,

yes, it seems in the FragmentEntryLinkEditorConfigContributor.java the toolbar is missing,
but in the FragmentEntryLinkRichTextEditorConfigContributor.java we have the toolbar and the issue is related to the RichTextEditor where the case is: clicking on an item in the ItemSelectorDialog not calls the selectedItemChange event. The event is only called when the Dialog is closed.

Could you please check this again?


Thanks, Roland

**Comment by dacousalr** [comment]
Hi @rolandpakai,

Have you tested the proposed change in https://github.com/dacousalr/liferay-portal/pull/193#pullrequestreview-376181895 ?

Applying it, the items can be selected when they are clicked.

**Comment by rolandpakai** [comment]
Hi @dacousalr 
sorry, but the proposed change in not realy clear for me.

In the FragmentEntryLinkEditorConfigContributor class i should change the following:
- remove the documentBrowseLinkUrl, filebrowserImageBrowseLinkUrl, filebrowserImageBrowseUrl from the jsonObject
- Define toolbars - What items should i define in the toolbar?

Thanks, Roland

**Comment by dacousalr** [comment]
Sorry if I am not being clear enough, @rolandpakai,

I will recap the conclusions:
- Conclusion 1: replace `eventName: editor.name + 'selectDocument',` with `eventName: editorName + 'selectItem',`. This (along the changes proposed in this PR) will make the events work as expected.
- Conclusion 2: remove documentBrowseLinkUrl, filebrowserImageBrowseLinkUrl, filebrowserImageBrowseUrl from the jsonObject in FragmentEntryLinkEditorConfigContributor. They are not needed since editable fields of type text do not have any toolbar. 

As a consequence, the FragmentEntryLinkRichTextEditorConfigContributor does not have to be changed with respect to this PR.
.
I hope this is clear enough. If not, tell me and I will send the PR.

**Comment by dacousalr** [comment]
Continue in https://github.com/rolandpakai/liferay-portal/pull/17

**Pull request closed without merging**